### DISCUSSION TITLE
Is support for  __round__() missing at the moment?

### DISCUSSION THREAD

**Discussion by shovnyk** [micropython/micropython#13069] [Core Development] [answered]
Hi. I was writing my own class for an RGB color library and decided to define the `__round__()` special function to convert fractional RGB values into the nearest integer. Having `round()` implicitly do that would be convenient. But it does not appear to work and I get the following error:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't convert ColorRGB to float
```

Admittedly this is quite trivial and I could very well just write another method for doing this. But I was just wondering if there are plans to add this.

**Comment by GitHubsSilverBullet** [comment]
An RGB value contain three distinct values for the three named colors.
What esoteric function would combine those three into a single meaningless float value?

**Reply by shovnyk** [reply]
I wasn't intending to return a float. I wanted to return another RGB object with the values of R, G and B rounded to the nearest integer in the range [0, 255] (because some methods in my class perform computations that make them fractional from time to time). Something like this (in CPython):
```
class ColorRGB:
   def __init__(self, r, g, b):
      self.r, self.g, self.b = r, g, b
      ...
   def __round__(self) -> ColorRGB:
      return ColorRGB(*[max(0, min(255, round(value))) for value in self])
```
But as @jimmo explained this is not possible as round calls `__float__` instead. That's fine. I'll just use another method instead for this.

**Comment by andrewleech** [comment]
Did you try adding __float__ function to your class? That's the normal way to allow conversion of a class to float.

Once that's there, the round function should work. I'm actually not sure if __round__ functionality is supported or not.

**Reply by jimmo** [reply, marked as answer]
@shovnyk `__round__` is not supported. This appears to be a deliberate decision, see e.g. https://github.com/micropython/micropython/issues/3610

And @andrewleech is correct, `round(x)` implicitly does `float(x)` which will call `x.__float__`.
